### DISCUSSION TITLE
Actions limitation, caching api tokens

### DISCUSSION THREAD

**Discussion by vozdelib** [zitadel/zitadel#8589] [General]
Hi, I'm investigating Zitadel as a potential replacement of Auth0 for our project.
I would like to ask about actions. Is there a way to cache some common data during action execution? For example, I need to call an external api, for that I need to first grab a token. I would like to cache it for half an hour and reuse it with other action executions. Is that possible anyhow? Something like this: https://auth0.com/blog/actions-caching-is-now-available/

Another question about any limitations the actions may have? Is there any size limitation for example? The input window for the action seems very small, indicating you expect not that much code in actions. Again I point to auth0 limitations: https://auth0.com/docs/customize/actions/limitations
they have a lot of limits. I imagine Zitadel might have some as well, but I can't find them listed anywhere.

Thank you.
Libor

**Comment by hifabienne** [comment]
At the moment we do not offer caching in the actions.
But we also need to differ between actions v1, and action v2. In the new version, the actions are more like a webhook where you run the code on your side. so you can implement caching if needed.

Actions V1: https://zitadel.com/docs/apis/actions/introduction
Actions v2: https://zitadel.com/docs/apis/actionsv2/introduction

@stebenz @adlerhurst can you help with the limits?

**Reply by vozdelib** [reply]
wow thanks, I haven't noticed this.
